### PR TITLE
Add ui info api

### PR DESCRIPTION
**Pull request by futuretea**
Signed-off-by: futuretea <Hang.Yu@suse.com>

**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

- pack ui plugin tar ball `https://releases.rancher.com/harvester-ui/plugin/harvester-latest.tar.gz` to harvester docker image
- new setting `ui-plugin-index` default to `https://releases.rancher.com/harvester-ui/plugin/harvester-latest/harvester-latest.umd.min.js`
- new api `/v1/harvester/ui-info` expose ui and plugin ui info
- new api `/v1/harvester/plugin-assets` serve bundled harvester ui plugin

**Related Issue:**
https://github.com/harvester/harvester/issues/2654

**Test plan:**

replace harvester image to `futuretea/harvester:ui-info-head`

1. view api `/v1/harvester/ui-info`
image

2. view api `/v1/harvester/plugin-assets/`
image

### CODE REVIEW

File: pkg/server/router.go [hunk 1]
```diff
@@ -46,6 +47,10 @@ func (r *Router) Routes(h router.Handlers) http.Handler {
 	kcGenerateHandler := kubeconfig.NewGenerateHandler(r.scaled, r.options)
 	m.Path("/v1/harvester/kubeconfig").Methods("POST").Handler(kcGenerateHandler)
 
+	uiInfoHandler := uiinfo.NewUIInfoHandler(r.scaled, r.options)
+	m.Path("/v1/harvester/uiinfo").Methods("GET").Handler(uiInfoHandler)
```
Comment (guangbochen) [inline pkg/server/router.go]: perfer to use `/s/uiinfo/ui-info/`

File: package/Dockerfile [hunk 2]
```diff
@@ -32,6 +34,9 @@ RUN mkdir -p /usr/share/harvester/harvester && \
     mkdir -p /usr/share/harvester/harvester/api-ui && \
     cd /usr/share/harvester/harvester/api-ui && \
     curl -sL https://releases.rancher.com/api-ui/${HARVESTER_API_UI_VERSION}.tar.gz | tar xvzf - --strip-components=1 && \
+    mkdir -p /usr/share/harvester/harvester/v1/harvester/plugin-assets && \
+    cd /usr/share/harvester/harvester/v1/harvester/plugin-assets && \
+    curl -sL https://releases.rancher.com/harvester-ui/plugin/harvester-${HARVESTER_UI_PLUGIN_BUNDLED_VERSION}.tar.gz | tar xvzf - --strip-components=1 && \
```
Comment (guangbochen) [inline package/Dockerfile:39]: @WuJun2016 do we need to keep the original UI assets, or only using the plugin UI would be enough?

File: pkg/api/uiinfo/ui_info_handler.go [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+			uiSource = "external"
+		} else {
+			uiSource = "bundled"
+		}
+	}
+	util.ResponseOKWithBody(rw, map[string]string{
+		settings.UISourceSettingName:               uiSource,
+		settings.UIIndexSettingName:                settings.UIIndex.Get(),
+		settings.UIPluginIndexSettingName:          settings.UIPluginIndex.Get(),
+		settings.UIPluginBundledVersionSettingName: os.Getenv(settings.GetEnvKey(settings.UIPluginBundledVersionSettingName)),
```
Comment (guangbochen) [inline pkg/api/uiinfo/ui_info_handler.go:36]: wondering what this setting response is used for?

Comment (futuretea) [reply pkg/api/uiinfo/ui_info_handler.go:36]: cc @WuJun2016

File: pkg/server/router.go (hunk 1 above)
Comment (futuretea) [reply pkg/server/router.go]: updated

File: pkg/api/uiinfo/ui_info_handler.go (hunk 3 above)
Comment (WuJun2016) [reply pkg/api/uiinfo/ui_info_handler.go:36]: The front-end needs to know what version is saved inside plugin-assets.  I know that normally the version of the plugin is the same as the harvester version, so its value should be the same as the `server-version`, but I'm not sure if using the `server-version` value is always appropriate.    
So to make sure I can always get the right value, so I added a property to save the version of the plugin (this plugin version is determined by the name of the packaged file)

If you don't add this attribute, then you need to expose the `server-version` to ui-info.

Comment (WuJun2016) [reply pkg/api/uiinfo/ui_info_handler.go:36]: I tested it, the `server-version` user is able to change, which is actually unreasonable.   
if the user changes this value, the user will not be able to access the Harvester from multiple clusters

File: package/Dockerfile (hunk 2 above)
Comment (WuJun2016) [reply package/Dockerfile:39]: We still need the old address, which is for a single cluster UI, and the newly added address for multiple clusters

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Review by bk201 — APPROVED** [review]

**Pull request merged**